### DISCUSSION TITLE
Unifi Protect - Change Recording Options

### DISCUSSION THREAD

**Discussion by gil04** [NRCHKB/node-red-contrib-unifi-os#64] [Q&A] [answered]
Hello all,

I would like to be able to configure the recording options of my unifi cameras via node-red.
For example, I would like to enable or disable recording.

Is this possible? And if so, can anyone give me a hint/tip on how to configure this?

Thx!

**Comment by crxporter** [comment, marked as answer]
This should definitely be possible. 

Looking back through our Discord channel - I expect it to be sent as a `PATCH` command from the request node.
![unknown](https://github.com/NRCHKB/node-red-contrib-unifi-os/assets/38265886/caa03349-df5a-40bf-ba10-2d3fb4f523b7)

This is to turn on (or off) the camera LED.

The important parts to change are the Endpoint which will be the same but with your camera ID (can get this from the bootstrap endpoint) and the data.

This will change to never record:
```json
{"recordingSettings": {"mode": "never"}}
```

Or always record:
```json
{"recordingSettings": {"mode": "always"}}
```

Or record detections only:
```json
{"recordingSettings": {"mode": "detections"}}
```

I just tested one of my cams and this works immediately to change the setting between always / never / detections only. Individual camera setting changes will always be this type format - to figure out another one you'll just need to make the settings change on unifi then compare the JSON response from Bootstrap for the camera... that will lead you to which part of the JSON changes when you adjust the settings.

**Reply by gil04** [reply]
Thanks a lot!!

That works right away!
